cal/cal/static/js/app.js

```diff
@@ -111,29 +111,41 @@ analyticsApp.component('tags', {
 function TagsDetailCtrl($scope, $http) {
   var tagUrl = '/v1/tags/' + this.tagId + '/events';
   var eventweek = '/v1/tags/' + this.tagId + '/eventWeek';
+  var query_timezone = moment.tz.guess();
   $scope.tagDetails = [];
   $scope.tagEvents = [];
+  $scope.tagHours = this.tagHours;
```

Comment:
For these three variables, change it to `this.tagDetails` etc, binding to the controller. Angular is moving away from the usage of scope for a variety of reasons such as it makes inheritance confusing, it pollutes the global namespace. After you've bound to your controller, in your html you can refer to it as `$ctrl.tagDetails`, or `TagsDetailCtrl.tagDetails` if you haven't used the `controllerAs` configuration.

[Link](https://toddmotto.com/no-scope-soup-bind-to-controller-angularjs/)
[Link 2](http://www.technofattie.com/2014/03/21/five-guidelines-for-avoiding-scope-soup-in-angular.html)